doc/tests/kernel/kernel_plan.rst

```diff
@@ -0,0 +1,117 @@
+.. _kernel_plan:
+
+Zephyr Kernel Verification Plan
+###############################
+
+Zephyr Test Overview
+********************
+
+To ensure robustness of the Zephyr Kernel, the Zephyr source code incorporates
+test cases using the Ztest framework. Zephyr verification mainly focuses on
+API correctness and feature completeness. Zephyr kernel testing uses different
+verification levels depending on need, as described below.
+
+* **API verification**
+
+    * Verify kernel objects initialization in all supported ways:
+
+        * K\_\*\_DEFINE (example: K\_THREAD\_STACK\_DEFINE, K\_STACK\_DEFINE)
+        * K\_\*\_INITIALIZER (example: \_K\_WORK\_INITIALIZER)
+        * k\_\*\_init (example: k\_sem\_init, k\_mutex\_init)
+
+    * Verify usages mentioned in Zephyr kernel API documentation:
+      :ref:`kernel_apis`
+
+    * Verify every API’s functionality with its typical parameters defined in
+      zephyr/kernel/include.
+
+    * Verify API invocation from each callable context (ISR, cooperative
+      thread and preemptive thread).
+
+    * Verify possible API calling sequence and/or normal state transition.
+
+    * API sequence test is to verify a kernel feature by calling its APIs
+      in various sequences. As an example of LIFO test:
+
+        * Init, put, put, get, get
+        * Init, get, get, put, put
+
+    * State transition test is to verify the kernel feature by calling its
+      APIs that triggering internal state change.
+      As an example of a thread’s life cycle:
+
+        * Spawn, delay start, suspend, resume, and abort
```

Comment:
Is "delay start" a part of a thread's life cycle, or are these two stages so it should be written "delay, start"?

Reply:
Should be "delayed start". I will make changes. Thanks